### PR TITLE
First draft of Dancer2::Plugin::Minion article

### PR DESCRIPTION
**Pull request by cromedome**
This article turned out to be less that what I had originally envisioned. So much of the reasoning for its existence was well covered in 2018, and this article really just turned into a bit more explanation of how to use the plugin to incorporate Minion instead of rolling your own thing.

Should I cover the need for having a job runner, or the process for building it? Is the code example clear enough? Are there any topics I haven't covered but should? @jberger, any feedback?

Thanks in advance!

### CODE REVIEW

**Comment by jberger** [comment]
It is a little spartan. Probably you should mention how to run the workers as that's the other side of the job that they'll need. Meanwhile having a slightly more tangible example would help. If you can't come up with a working one then at least a motivation, like generating thumbnails on uploading a full size image or generating a PDF from a page of content.

**Comment by jberger** [comment]
Also, is there any way to see the minion dashboard?

**Comment by cromedome** [comment]
Major update! Please review.

File: pending/20-dancer2-plugin-minion.pod [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+Two more keywords are created by the plugin, C<add_task> and C<enqueue>, that map directly 
+to the same methods available in C<Minion>. These tasks are common enough to warrant having
+their own keywords to save you the little bit of additional typing to use them.
+
+Lastly, the keyword C<minion_app> creates a Minion application, which you can then mount
+via L<Plack> (if you want to have an admin dashboard for Minion in your Dancer app), or 
+if you want to enable the Minion CLI in your Dancer apps. I will demonstrate this below.
+
+Here's a totally fabricated example to demonstrate how each of these methods can be
```
Comment (jberger) [inline pending/20-dancer2-plugin-minion.pod:43]: so now that you have the introduction to the example, you don't need this line, 
it should probably just say something like "lets see what that looks like" or so

File: pending/20-dancer2-plugin-minion.pod [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    };
+
+Here, we consume a file upload, stash it on disk, and call the C<thumbnails>
+job to render multiple thumbnails. This might take a while, so we don't want
+to tie up the browser waiting for them to finish.
+
+    builder {
+        # mount the container app at /dashboard/
+        # note that the trailing slash is very important
+        mount '/dashboard/' => minion_app( '/foo/bar' )->start;
```
Comment (jberger) [inline pending/20-dancer2-plugin-minion.pod:98]: This line needs an absolute url

File: pending/20-dancer2-plugin-minion.pod [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        mount '/' => start;
+    };
+
+Finally, we create the L<Plack> application by mounting our L<Dancer2> 
+application to the root URL, and attaching the L<Minion> dashboard to the
+C</dashboard/> URL (the trailing C</> is required!). It also sets the 
+"Back to Site" link on the dashboard to the ficticious URL C</foo/bar>.
```
Comment (jberger) [inline pending/20-dancer2-plugin-minion.pod:105]: so does this one

File: pending/20-dancer2-plugin-minion.pod [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        my $target = path('public', $name);
+        $file->copy_to($target);
+        enqueue(thumbnails => [$target]);
+        redirect "/$name";
+    };
+
+    builder {
+        # mount the container app at /dashboard/
+        # note that the trailing slash is very important
+        mount '/dashboard/' => minion_app( 'http://localhost:5000/upload' )->start;
```
Comment (jberger) [inline pending/20-dancer2-plugin-minion.pod:152]: and this absolute url, while probably correct for the demo isn't really what users are going to think of as a url, I'd use example.com or northpole.org or something

File: pending/20-dancer2-plugin-minion.pod (hunk 1 above)
Comment (cromedome) [reply pending/20-dancer2-plugin-minion.pod]: Done!

File: pending/20-dancer2-plugin-minion.pod (hunk 2 above)
Comment (cromedome) [reply pending/20-dancer2-plugin-minion.pod]: Done :)

File: pending/20-dancer2-plugin-minion.pod (hunk 3 above)
Comment (cromedome) [reply pending/20-dancer2-plugin-minion.pod]: Also done! :)

File: pending/20-dancer2-plugin-minion.pod (hunk 4 above)
Comment (cromedome) [reply pending/20-dancer2-plugin-minion.pod]: Done :)

**Comment by cromedome** [comment]
Merged!

**Pull request closed without merging**